src/main/java/org/zalando/nakadi/service/job/ExclusiveJobWrapper.java

```diff
@@ -0,0 +1,132 @@
+package org.zalando.nakadi.service.job;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Charsets;
+import org.apache.curator.utils.ZKPaths;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.zalando.nakadi.repository.zookeeper.ZooKeeperHolder;
+
+/**
+ * Class helps to run jobs that should be performed periodically only on single Nakadi instance
+ */
+public class ExclusiveJobWrapper {
+
+    public static final String NAKADI_JOBS_PATH = "/nakadi/jobs";
+
+    private final Logger log;
+    private final String zkPath;
+    private final ZooKeeperHolder zkHolder;
+    private final ObjectMapper objectMapper;
+    private final long jobPeriodMs;
+
+    public ExclusiveJobWrapper(final ZooKeeperHolder zkHolder,
+                               final ObjectMapper objectMapper,
+                               final String jobName,
+                               final long jobPeriodMs) {
+        this.zkHolder = zkHolder;
+        this.objectMapper = objectMapper;
+        this.jobPeriodMs = jobPeriodMs;
+        this.zkPath = ZKPaths.makePath(NAKADI_JOBS_PATH, jobName);
+        this.log = LoggerFactory.getLogger("nakadi-job." + jobName);
+    }
+
+    public void runJobLocked(final Runnable action) {
+        log.info("Trying to run job...");
+
+        if (!timeToRunAction()) {
```

Comment:
It is done like that because the job can be executed on any of Nakadi instances and Nakadi instances have different start time so they have different timers for this Job.
So if I would implement it just to run each N minutes - then the job can be executed just few seconds after it was executed on another Nakadi instance - so the running of Job will be quite random. That's why I need this check if it's time to run job (it checks the timestamp when the Job was executed the last time)